airflow/models/xcom.py

```diff
@@ -97,7 +97,14 @@ def set(cls, key, value, task_id, dag_id, execution_date=None, run_id=None, sess
 
             execution_date = dag_run.execution_date
 
-        value = XCom.serialize_value(value)
+        value = XCom.serialize_value(
+            value=value,
+            key=key,
+            task_id=task_id,
+            dag_id=dag_id,
+            execution_date=execution_date,
+            run_id=run_id,
+        )
```

Comment:
This counts as a breaking change -- an existing custom XCom backend would likely break as a result, so we can't do this the "simple" way.

We'd have to look at the function signature and check before calling it.

Reply:
@ashb with that in mind, for the _new_ signature, do you think `serialize_value` should just take `kwargs`, or should we add all these params?

Reply:
Not sure -- haven't thought about it in detail.

Probably should be forced to use named args, (but not just a `**kwargs` catch all).

We should bear AIP-42 in mind here too, which might add some form of `mapping_index` argument.